utils/common/src/main/kotlin/DirectoryStash.kt

```diff
@@ -32,24 +32,34 @@ import kotlin.io.path.moveTo
 import org.apache.logging.log4j.kotlin.Logging
 
 /**
- * A convenience function that stashes directories using a [DirectoryStash] instance.
+ * A convenience function that stashes [directories] to the [stashParentDir] using a [DirectoryStash] instance. If
+ * [deleteParentOnClose] is true, the [stashParentDir] is expected to be empty on close and will be deleted.
  */
-fun stashDirectories(vararg directories: File): Closeable = DirectoryStash(setOf(*directories))
+fun stashDirectories(
+    vararg directories: File,
+    stashParentDir: File = createTempDirectory("stashParent").toFile(),
+    deleteParentOnClose: Boolean = true
+): Closeable = DirectoryStash(setOf(*directories), stashParentDir, deleteParentOnClose)
 
 /**
- * A Closable class which temporarily moves away directories and moves them back on close. Any conflicting directory
- * created at the location of an original directory is deleted before the original state is restored. If a specified
- * directory did not exist on initialization, it will also not exist on close.
+ * A [Closeable] class which temporarily moves away all [directories] to inside an existing [stashParentDir] and moves
+ * them back on close. If [deleteParentOnClose] is true, the [stashParentDir] is expected to be empty on close and will
+ * be deleted. Any conflicting directory created at the location of an original directory is deleted before the original
+ * state is restored. If a specified directory did not exist on initialization, it will also not exist on close.
  */
-private class DirectoryStash(directories: Set<File>) : Closeable {
+private class DirectoryStash(
+    directories: Set<File>,
+    private val stashParentDir: File,
+    private val deleteParentOnClose: Boolean
+) : Closeable {
     private companion object : Logging
 
     private val stashedDirectories: Map<File, File?> = directories.associateWith { originalDir ->
         // We need to check this on each iteration instead of filtering beforehand to properly handle parent / child
         // directories.
         if (originalDir.isDirectory) {
             // Create a temporary directory to move directories as-is into.
-            val tempDir = createTempDirectory(originalDir.parentFile.toPath(), "stash").toFile()
+            val tempDir = createTempDirectory(stashParentDir.toPath(), "stash").toFile()
```

Comment:
Do we have the requirement now that `stashParentDir` exists?

Reply:
This became irrelevant due to the new approach.